status/k8s.go

```diff
@@ -172,7 +172,11 @@ func (k *K8sStatusCollector) podCount(ctx context.Context, status *Status) error
 
 	ciliumEps, err := k.client.ListCiliumEndpoints(ctx, "", metav1.ListOptions{})
 	if err != nil {
-		return err
+		// When the CEP has not been registered yet, it's impossible
+		// for any pods to be managed by Cilium.
+		if err.Error() != "the server could not find the requested resource (get ciliumendpoints.cilium.io)" {
```

Comment:
Errors in Go should not be asserted by string matching, this is likely to lead to bugs when error strings change. Instead the `errors.As` (https://pkg.go.dev/errors#As) method can be used to search for a given error type in the wrapped error tree. 

In this case `http.StatusNotFound` is the error type that you want to assert with `errors.As`.

https://github.com/kubernetes/apimachinery/blob/master/pkg/api/errors/errors.go#L446C7-L446C26

You can also move this predicate up into line 174 to avoid an extra level of indentation and simplify things. That is:

```
if err != nil and !errors.As(err, http.StatusNotFound) {
	return err
}
```

Reply:
I fully agree that error string matching is a bad approach.
But I don't think we can use `errors.As` because we don't have error wrapping here.

So, I used type assertion and status code check.